hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystem.java

```diff
@@ -982,6 +985,19 @@ public boolean exists(Path f) throws IOException {
     return super.exists(f);
   }
 
+  @Override
+  public RemoteIterator<FileStatus> listStatusIterator(Path path)
+      throws IOException {
+    LOG.debug("AzureBlobFileSystem.listStatusIterator path : {}", path);
+    if (abfsStore.getAbfsConfiguration().enableAbfsListIterator()) {
```

Comment:
would you ever want to make this not optional? one code path == better testing

Reply:
after it is used for a while and we have confidence that it is baked

Reply:
OK, but it's something to plan to cut after a release has been out. So file a new JIRA about cutting the old one.

Reply:
Done
https://issues.apache.org/jira/browse/HADOOP-17512